### PR TITLE
refactor(promises): Add promise linting and fix current errors

### PR DESCRIPTION
**Pull request by melink14**
- Add recommended lint warnings to config
- Prefer await over promise.then

### CODE REVIEW

File: extension/data.ts [hunk 1]
```diff
@@ -122,21 +122,30 @@ class RcxDict {
...
+    const loadDeinflectionData = this.loadDeinflectionData();
+    let loadNameData = Promise.resolve('');
+    let loadNameIndex = Promise.resolve('');
     if (loadNames) {
-      this.fileReadAsync(chrome.extension.getURL('data/' + 'names.dat')).then(
-        (data) => {
-          this.nameDict = data;
-        }
+      loadNameData = this.fileReadAsync(
+        chrome.extension.getURL('data/' + 'names.dat')
```
Comment (Stephie) [inline extension/data.ts:131]: `'data/names.dat'` (here and elsewhere)

File: extension/data.ts [hunk 2]
```diff
@@ -122,21 +122,30 @@ class RcxDict {
   async init(loadNames: boolean) {
     const started = +new Date();
 
-    await this.loadDictionaries();
-    await this.loadDeinflectionData();
+    const loadDictionaries = this.loadDictionaries();
+    const loadDeinflectionData = this.loadDeinflectionData();
```
Comment (Stephie) [inline extension/data.ts:126]: Inline these in `Promises.all()`

**Review by Stephie — CHANGES_REQUESTED** [review]

**Review by Stephie — APPROVED** [review]

File: extension/data.ts (hunk 2 above)
Comment (melink14) [reply extension/data.ts:126]: Done!

File: extension/data.ts (hunk 1 above)
Comment (melink14) [reply extension/data.ts:131]: done!

File: extension/data.ts [hunk 3]
```diff
@@ -116,23 +116,19 @@ class RcxDict {
...
-        }
-      );
-      this.fileReadAsync(chrome.extension.getURL('data/' + 'names.idx')).then(
-        (data) => {
-          this.nameIndex = data;
-        }
-      );
-    }
+    // await all loading promises in parallel, while also saving the values of
+    // the name dictionaries into their proper place.
```
Comment (Stephie) [inline extension/data.ts:123]: The code might be self-explanatory enough.

**Review by Stephie — COMMENTED** [review]
So clean! 🤩

**Pull request merged**